tests/components/zha/test_init.py

```diff
@@ -311,3 +315,44 @@ async def test_timezone_update(
 
     assert hass.config.time_zone == "America/New_York"
     assert gateway.config.local_timezone == zoneinfo.ZoneInfo("America/New_York")
+
+
+async def test_setup_firmware_update_in_progress(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test that ZHA setup is blocked when firmware update is in progress."""
+    await async_setup_component(hass, "homeassistant_hardware", {})
+
+    config_entry.add_to_hass(hass)
+    device_path = config_entry.data[CONF_DEVICE][CONF_DEVICE_PATH]
+
+    # Register a firmware update in progress for the device
+    async_register_firmware_update_in_progress(hass, device_path, "skyconnect")
+
+    # ZHA setup should fail and the config entry should be in retry state
+    await hass.config_entries.async_setup(config_entry.entry_id)
+
+    # Verify the config entry is not loaded and is in setup retry state
+    assert config_entry.state.name == "SETUP_RETRY"
```

Comment:
We normally compare with the state enum.

https://github.com/home-assistant/core/blob/20949d39c45ea7add2fa6a2f0132f49f486f1d5c/homeassistant/config_entries.py#L141